db/appsock_handler.c

```diff
@@ -265,6 +268,33 @@ static void appsock_work(struct thdpool *pool, void *work, void *thddata)
     struct appsock_thd_state *state = thddata;
     appsock_work_args_t *w = (appsock_work_args_t *)work;
     int keepsocket = 0;
+    int curr_appsock_conns = 0;
+    w->rejected = 0;
+    int num_retry = 0;
+
+retry:
+    Pthread_mutex_lock(&appsock_conn_lk);
+    if (active_appsock_conns >= bdb_attr_get(thedb->bdb_attr, BDB_ATTR_MAXAPPSOCKSLIMIT)) {
+        if (close_lru_client(active_appsock_conns, NULL, bdb_attr_get(thedb->bdb_attr, BDB_ATTR_MAXAPPSOCKSLIMIT)) != 0) {
+            num_retry++;
+            if (num_retry < 5) {
+                Pthread_mutex_unlock(&appsock_conn_lk);
+                sleep(1);
+                goto retry; 
+            }
+            w->rejected = 1; /* Reject this query. */
```

Comment:
The only reason for rejected seems to be to force-allow admin connections.  We already know at this point that whether it's admin - we can skip this entire process for admin, and reject non-admin connections right at this point.

Reply:
Yes, we have to give error back based on plugin